jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/vminstance/VMInstanceSSHKeySecurityFeatureRule.java

```diff
@@ -0,0 +1,130 @@
+
+        if (!StringUtils.isNullOrEmpty(resourceId)) {
+
+            Map<String, Object> mustFilter = new HashMap<>();
+            mustFilter.put(PacmanUtils.convertAttributetoKeyword(PacmanRuleConstants.RESOURCE_ID), resourceId);
+            mustFilter.put(PacmanRuleConstants.LATEST, true);
+
+            try {
+                isBlockProjectWideSSHEnable = verifyIsBlockProjectWideSSHEnable(vmEsURL, mustFilter);
+                if (!isBlockProjectWideSSHEnable) {
+                    List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                    LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+
+                    annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                    annotation.put(PacmanSdkConstants.DESCRIPTION, "Block Project-Wide SSH Keys is not enabled");
+                    annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                    annotation.put(PacmanRuleConstants.CATEGORY, category);
+
+                    issue.put(PacmanRuleConstants.VIOLATION_REASON, "Block Project-Wide SSH Keys is not enabled");
+                    issueList.add(issue);
+                    annotation.put("issueDetails", issueList.toString());
+                    logger.debug("========VMInstanceSSHKeySecurityFeatureRule ended with an annotation {} : =========", annotation);
+                    return new RuleResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE, annotation);
+                }
+
+            } catch (Exception exception) {
+                throw new RuleExecutionFailedExeption(exception.getMessage());
+            }
+        }
+
+        logger.debug("========VMInstanceMigrateOnMaintainence ended=========");
+        return new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+
+    }
+
+    private boolean verifyIsBlockProjectWideSSHEnable(String vmEsURL, Map<String, Object> mustFilter) throws Exception {
+        logger.debug("========verifyIsBlockProjectWideSSHEnable started=========");
+        JsonArray hitsJsonArray = GCPUtils.getHitsArrayFromEs(vmEsURL, mustFilter);
+        boolean validationResult = false;
+        if (hitsJsonArray.size() > 0) {
+            JsonObject vmInstanceObject = (JsonObject) ((JsonObject) hitsJsonArray.get(0))
+                    .get(PacmanRuleConstants.SOURCE);
+
+            logger.debug("Validating the data item: {}", vmInstanceObject);
+            JsonArray items=vmInstanceObject.getAsJsonObject().get(PacmanRuleConstants.ITEMS).getAsJsonArray();
+
+            if(!items.isEmpty()){
+                for(int i=0;i<items.size();i++){
+                    JsonObject item=((JsonObject) items.get(i));
+
+                    String key = item.getAsJsonObject()
+                            .get(PacmanRuleConstants.KEY).getAsString();
+
+                    if(key.equalsIgnoreCase(keyValue)) {
```

Comment:
We should also validate the value present against key block-project-ssh-keys. It should be true(on) for compliant resources

Reply:
Added the value flag to validate  against key "block-project-ssh-keys".